maml_rl/samplers/multi_task_sampler.py

```diff
@@ -304,35 +328,67 @@ def create_episodes(self,
+            prev_triplets, chosen_actions, prev_game_facts = [], [], []
+            prev_step_dones, prev_scores = [], []
+            for _ in range(self.batch_size):
+                prev_triplets.append([])
+                chosen_actions.append('tw-restart')
+                prev_game_facts.append(set())
+                prev_step_dones.append(0.0)
+                prev_scores.append(0.0)
+            ####
+            # Don't need Rl^2 here but just in case
+            meta_dones = to_pt(np.zeros(self.batch_size), enable_cuda=self.agent.use_cuda, type='float')
+            meta_torch_step_rewards = to_pt(np.zeros(self.batch_size), enable_cuda=self.agent.use_cuda, type='float')
+            meta_prev_h = to_pt(np.zeros((1, self.batch_size, self.agent.policy_net.block_hidden_dim)), enable_cuda=self.agent.use_cuda, type='float')
+            ####
             while not self.envs.dones.all():
+                print(infos["infos"][0]["admissible_commands"])
                 observations = [info["feedback"] for info in infos["infos"]]
-
+                info_for_agent = [info for info in infos["infos"]]
+                print("LENG : " + str(len(info_for_agent)))
+                observation_strings, current_triplets, action_candidate_list, dict_info_for_agent, _, current_game_facts = self.agent.get_game_info_at_certain_step_maml(info_for_agent, prev_actions=chosen_actions, prev_facts=None)
+                observation_strings = [item + " <sep> " + a for item, a in zip(observation_strings, chosen_actions)]
+                value, chosen_actions, meta_prev_h, action_log_probs, chosen_indices, _, prev_h, prev_c = self.agent.act(observation_strings, current_triplets, action_candidate_list, meta_dones.unsqueeze(1), meta_torch_step_rewards.unsqueeze(1), meta_prev_h)
+                chosen_actions = [(action if not done else "restart") for done, action in zip(dones, chosen_actions)]
+                chosen_actions_before_parsing = [(item[idx] if not done else "*restart*") for item, idx, done in zip(dict_info_for_agent["admissible_commands"], chosen_indices, dones)]
+                
+                ######
                 # TODO:
-                # observations_tensor = torch.from_numpy(observations)
+                # observations_tensor = torch.from_numpy(observations) ## Do we realy want numpy? If so, i will need to demarcate inside the agent.act and essentialy write the function explicitly write here--easy
                 # pi = self.policy(observations_tensor, params=params)
                 # actions_tensor = pi.sample()
                 # actions = actions_tensor.cpu().numpy()
-                actions = ["look"] * self.batch_size
+                # actions = ["look"] * self.batch_size
+                #actions = chosen_actions_before_parsing.cpu().numpy()
 
-                new_observations, rewards, _, infos = self.envs.step(actions)
+                new_observations, rewards, dones, infos = self.envs.step(chosen_actions_before_parsing)
```

Comment:
Here, `new_observations` is a numpy array of zeros. You should use the observations obtained from `infos` (like done above).

Reply:
For reference: 
https://github.com/Ashutosh-Adhikari/pytorch-maml-rl/blob/tw-maml/maml_rl/envs/textworld.py#L139

Reply:
i see

Reply:
actually it's been overridden anyway